iocore/net/SSLNetVConnection.cc

```diff
@@ -1747,6 +1747,7 @@ SSLNetVConnection::callHooks(TSEvent eventId)
   }
 
   if (curHook != nullptr) {
+    SCOPED_MUTEX_LOCK(lock, curHook->m_cont->mutex, this_ethread());
```

Comment:
Do you mean there is a case of `curHook->m_cont->mutex` is `nullptr` by "Requires #5857".

Reply:
No, but overall the code is simplified by putting the check in `SCOPED_MUTEX_LOCK`. Otherwise every place that invokes the continuation would have to have the same cut and pasted logic. This is the reason that was moved in to `MUTEX_TRY_LOCK`.

Reply:
Yeah, I meant it needs the null checking , since some times the `mutex` could be pointing to nothing

Reply:
Less dup code is fine but failing silently doesn’t sounds great.

Reply:
I don't think it count as failing. If a mutex is provided, it locks it just the same as before, if a nullptr is provided nothing happens.

Reply:
If no mutex is present, I read that as a indication no locking is required. Because the goal is to not have an unlocked mutex, this is success because no mutex trivially means no unlocked mutex.

Reply:
> if a nullptr is provided nothing happens.
> If no mutex is present, I read that as a indication no locking is required.

Like I wrote on #5857, I don't like this behavior. Who expect the behavior?
When nullptr is passed to SCOPED_MUTEX_LOCK, are you always sure it's intentional?

Reply:
It's the behavior I would expect. If I set the mutex to `nullptr`, that says "I don't need locking" and I don't expect to get a block or crash because of it. Again, how should a continuation be marked to indicate "no locking required"?

Reply:
Did you read my comment on #5857? I suggested to have another macro for the behavior. Passing nullptr by mistake should be detected as assert failure, error, crash, etc.

Reply:
I agree with Maskit, we should have separate macros for doing the nullptr check and it should be moved into the macro instead of in the constructor of the class.  For most code it is OK to push down nullptr checks, but for a mutex you want to guarantee that you have the lock in a lot of cases.

I recommendation would be SCOPED_MUTEX_LOCK_IF_SET and MUTEX_TRY_LOCK_IF_SET.

Reply:
That doesn't help - this macro isn't accessible from the C API and so another macro doesn't provide the opportunity to specify "no lock needed". Passing `nullptr` isn't a mistake, even in your example it's done quite deliberately during creation of the continuation.

Perhaps you could be more explicit about an alternative that would allow having `nullptr` for a continuation mutex that does not result in an assertion based crash?

Reply:
Looking at the use of SCOPED_MUTEX_LOCK in most cases we don't do a nullptr check and require that there is a mutex.  This would change the semantics in those cases and if some part of the code accidentally sets the mutex to null (not uncommon that memory is accentually written to) we would then not be holding a mutex.

The issue that @maskit has should have been addressed in #5857 before approving and merging the change.